tests/phpunit/tests/block-supports/wpRenderDimensionsSupport.php

```diff
@@ -0,0 +1,176 @@
+		$GLOBALS['wp_theme_directories'] = array( WP_CONTENT_DIR . '/themes', $this->theme_root );
+
+		add_filter( 'theme_root', array( $this, 'filter_set_theme_root' ) );
+		add_filter( 'stylesheet_root', array( $this, 'filter_set_theme_root' ) );
+		add_filter( 'template_root', array( $this, 'filter_set_theme_root' ) );
+
+		// Clear caches.
+		wp_clean_themes_cache();
+		unset( $GLOBALS['wp_themes'] );
+		WP_Style_Engine_CSS_Rules_Store::remove_all_stores();
+	}
+
+	public function tear_down() {
+		$GLOBALS['wp_theme_directories'] = $this->orig_theme_dir;
+
+		// Clear up the filters to modify the theme root.
+		remove_filter( 'theme_root', array( $this, 'filter_set_theme_root' ) );
+		remove_filter( 'stylesheet_root', array( $this, 'filter_set_theme_root' ) );
+		remove_filter( 'template_root', array( $this, 'filter_set_theme_root' ) );
+
+		wp_clean_themes_cache();
+		unset( $GLOBALS['wp_themes'] );
+		WP_Style_Engine_CSS_Rules_Store::remove_all_stores();
+		unregister_block_type( $this->test_block_name );
+		$this->test_block_name = null;
+		parent::tear_down();
+	}
+
+	public function filter_set_theme_root() {
+		return $this->theme_root;
+	}
+
+	/**
+	 * Tests that dimensions block support works as expected.
+	 *
+	 * @ticket 60365
+	 *
+	 * @covers ::wp_render_dimensions_support
+	 *
+	 * @dataProvider data_dimensions_block_support
+	 *
+	 * @param string $theme_name          The theme to switch to.
+	 * @param string $block_name          The test block name to register.
+	 * @param mixed  $dimensions_settings The dimensions block support settings.
+	 * @param mixed  $dimensions_style    The dimensions styles within the block attributes.
+	 * @param string $expected_wrapper    Expected markup for the block wrapper.
+	 * @param string $wrapper             Existing markup for the block wrapper.
+	 */
+	public function test_dimensions_block_support( $theme_name, $block_name, $dimensions_settings, $dimensions_style, $expected_wrapper, $wrapper ) {
+		switch_theme( $theme_name );
+		$this->test_block_name = $block_name;
+
+		register_block_type(
+			$this->test_block_name,
+			array(
+				'api_version' => 2,
+				'attributes'  => array(
+					'style' => array(
+						'type' => 'object',
+					),
+				),
+				'supports'    => array(
+					'dimensions' => $dimensions_settings,
+				),
+			)
+		);
+
+		$block = array(
+			'blockName' => $block_name,
+			'attrs'     => array(
+				'style' => array(
+					'dimensions' => $dimensions_style,
+				),
+			),
+		);
+
+		$actual = wp_render_dimensions_support( $wrapper, $block );
+
+		$this->assertEquals(
+			$expected_wrapper,
+			$actual,
+			'Dimensions block wrapper markup should be correct'
+		);
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_dimensions_block_support() {
+		return array(
+			'aspect ratio style is applied, with min-height unset' => array(
+				'theme_name'          => 'block-theme-child-with-fluid-typography',
```

Comment:
Any reason for using this particular theme as opposed to, say, `block-theme-child`?

Reply:
Yes, `block-theme-child-with-fluid-typography` has `appearanceTools` set to `true` so that the block support will be in use. I think we're doing the same with the background image support, too.

Reply:
Oh I see! Might be worth consolidating some of those themes at some point but this is fine for now.